test/src/main/java/org/apache/accumulo/test/shell/ShellServerIT.java

```diff
@@ -1259,6 +1267,105 @@ public void goal() throws Exception {
+      assertTrue(result.contains("TABLET ID    HOSTING GOAL"));
+      assertTrue(result.contains("1;d<         ONDEMAND"));
+      assertTrue(result.contains("1;m;d        ONDEMAND"));
+      assertTrue(result.contains("1;s;m        ONDEMAND"));
+      assertTrue(result.contains("1<;s         ONDEMAND"));
+
+      ts.exec("setgoal -g ALWAYS -r p");
+      result = ts.exec("getgoal -r p");
+      assertFalse(result.contains("1;d<         ONDEMAND"));
+      assertFalse(result.contains("1;m;d        ONDEMAND"));
+      assertTrue(result.contains("1;s;m        ALWAYS"));
+      assertFalse(result.contains("1<;s         ONDEMAND"));
+
+      result = ts.exec("getgoal");
+      assertTrue(result.contains("1;d<         ONDEMAND"));
+      assertTrue(result.contains("1;m;d        ONDEMAND"));
+      assertTrue(result.contains("1;s;m        ALWAYS"));
+      assertTrue(result.contains("1<;s         ONDEMAND"));
+
+      result = ts.exec("getgoal -b f -e p");
+      assertFalse(result.contains("1;d<         ONDEMAND"));
+      assertTrue(result.contains("1;m;d        ONDEMAND"));
+      assertTrue(result.contains("1;s;m        ALWAYS"));
+      assertFalse(result.contains("1<;s         ONDEMAND"));
+
+    } finally {
+      if (splitsFilePath != null) {
+        Files.delete(splitsFilePath);
+      }
+    }
+  }
+
+  // Verify that when splits are added after table creation, hosting goals are set properly
+  @Test
+  public void testGetGoalCommand_DelayedSplits() throws IOException {
+
+    final String[] tableName = getUniqueNames(2);
+
+    ts.exec("createtable " + tableName[0], true);
+    String result = ts.exec("getgoal");
+    assertTrue(result.contains("TABLE: " + tableName[0]));
+    assertTrue(result.contains("TABLET ID    HOSTING GOAL"));
+    assertTrue(result.contains("1<<          ONDEMAND"));
+
+    // add the splits and check goals again
+    ts.exec("addsplits d m s", true);
+    result = ts.exec("getgoal");
+    assertTrue(result.contains("1;d<         ONDEMAND"));
+    assertTrue(result.contains("1;m;d        ONDEMAND"));
+    assertTrue(result.contains("1;s;m        ONDEMAND"));
+    assertTrue(result.contains("1<;s         ONDEMAND"));
+
+    // scan metadata table to be sure the hosting goals were properly set
+    result = ts.exec("scan -t accumulo.metadata -c hosting:goal", true);
+    assertTrue(result.contains("1;d hosting:goal []\tONDEMAND"));
+    assertTrue(result.contains("1;m hosting:goal []\tONDEMAND"));
+    assertTrue(result.contains("1;s hosting:goal []\tONDEMAND"));
+    assertTrue(result.contains("1< hosting:goal []\tONDEMAND"));
+
+    ts.exec("createtable " + tableName[1] + " -g always", true);
+    result = ts.exec("getgoal");
+    assertTrue(result.contains("TABLE: " + tableName[1]));
+    assertTrue(result.contains("TABLET ID    HOSTING GOAL"));
+    assertTrue(result.contains("2<<          ALWAYS"));
+
+    ts.exec("addsplits d m s", true);
+    result = ts.exec("getgoal");
```

Comment:
I don't think I have seen a test that uses the `-t` argument. Is the current test taking advantage of the current table context within the shell?